app/Run.hs

```diff
@@ -545,53 +504,240 @@ mkChainDbArgs TraceOptions {traceChainDB} cfg initLedger registry (CoreNodeId ni
+        ChainDB.InvalidBlock err pt -> tr $ WithTip tip $
           "Invalid block " <> condense pt <> ": " <> show err
         _ -> ignore
       _  -> ignore
-    ChainDB.TraceLedgerEvent ev -> case ev of
-      ChainDB.InitLog ev' -> traceInitLog ev'
-      ChainDB.TookSnapshot snap pt -> tr $
+    WithTip tip (ChainDB.TraceLedgerEvent ev) -> case ev of
+      ChainDB.InitLog ev' -> traceInitLog tip ev'
+      ChainDB.TookSnapshot snap pt -> tr $ WithTip tip $
         "Took ledger snapshot " <> show snap <> " at " <> condense pt
-      ChainDB.DeletedSnapshot snap -> tr $
+      ChainDB.DeletedSnapshot snap -> tr $ WithTip tip $
         "Deleted old snapshot " <> show snap
-    ChainDB.TraceCopyToImmDBEvent ev -> case ev of
-      ChainDB.CopiedBlockToImmDB pt -> tr $
+    WithTip tip (ChainDB.TraceCopyToImmDBEvent ev) -> case ev of
+      ChainDB.CopiedBlockToImmDB pt -> tr $ WithTip tip $
         "Copied block " <> condense pt <> " to the ImmutableDB"
       _ -> ignore
-    ChainDB.TraceGCEvent ev -> case ev of
-      ChainDB.PerformedGC slot       -> tr $
+    WithTip tip (ChainDB.TraceGCEvent ev) -> case ev of
+      ChainDB.PerformedGC slot       -> tr $ WithTip tip $
         "Performed a garbage collection for " <> condense slot
       _ -> ignore
-    ChainDB.TraceOpenEvent ev -> case ev of
-      ChainDB.OpenedDB immTip tip -> tr $
+    WithTip tip (ChainDB.TraceOpenEvent ev) -> case ev of
+      ChainDB.OpenedDB immTip tip' -> tr $ WithTip tip $
         "Opened with immutable tip at " <> condense immTip <>
-        " and tip " <> condense tip
+        " and tip " <> condense tip'
       _ -> ignore
     _ -> ignore
   where
-    tr = traceWith tracer
+    tr :: Show a => WithTip blk a -> m ()
+    tr = traceWith (contramap show tracer)
 
     ignore :: m ()
     ignore = return ()
 
-    traceInitLog = \case
-      LedgerDB.InitFromGenesis -> tr "Initialised the ledger from genesis"
-      LedgerDB.InitFromSnapshot snap tip -> tr $
+    traceInitLog :: Point blk -> LedgerDB.InitLog (Point blk) -> m ()
+    traceInitLog tip = \case
+      LedgerDB.InitFromGenesis -> tr $ WithTip tip "Initialised the ledger from genesis"
+      LedgerDB.InitFromSnapshot snap tip' -> tr $ WithTip tip $
         "Initialised the ledger from snapshot " <> show snap <> " at " <>
-        condense (tipToPoint tip)
+        condense (tipToPoint tip')
       LedgerDB.InitFailure snap _failure initLog -> do
-          tr $ "Snapshot " <> show snap <> " invalid"
-          traceInitLog initLog
+          tr $ WithTip tip $ "Snapshot " <> show snap <> " invalid"
+          traceInitLog tip initLog
+
+data Traces peer blk = Traces {
+    -- | by default we use 'readableChainDB' tracer, if on this it will use more
+    -- verbose tracer
+    --
+      tracerChainDB
+        :: !(Tracer IO(WithTip blk (ChainDB.TraceEvent blk)))
+
+    -- | consensus tracer
+    --
+    , tracerConsensus
+      :: !(Tracer IO String)
```

Comment:
"String" here is a red flag. Add a TODO.

Reply:
@mrBliss has just fixed it in `ouroboros-consensus`. It will be fixed today.

Reply:
I'll do that in a separate PR.